iter/iter_test.go

```diff
@@ -38,7 +37,7 @@ func TestIterator(t *testing.T) {
 		// iter.Concurrency > numInput case that updates iter.Concurrency
 		iterator.ForEachIdx([]int{1, 2, 3}, func(i int, t *int) {})
 
-		assert.Equal(t, iterator.MaxGoroutines, 999)
+		require.Equal(t, iterator.MaxGoroutines, 999)
```

Comment:
Not major but IMO this should be `assert.Equal`, not `require.Equal` - the latter should only be used for tests that need to fast-fail such that the rest of the test does not run, in this case there are no assertions after this one so there's no need to use `require`

Reply:
I agree that there's no _need_ to, but I'm also pretty convinced there's no real reason to use `assert`. I don't think I've ever wanted a test to fail but continue running. So I'm not sure I see what value using `assert` here adds either. Really, the only time I can picture it mattering is if we add more to this test, in which case I would probably want `require`, so it makes sense to just use `require` everywhere.

In what cases would you _not_ want a test to fast-fail?

Reply:
> In what cases would you not want a test to fast-fail?

It's useful when doing more complex assertions, where it's nice to know if some assertions continued to hold when others have failed, or exactly which of multiple assertions have failed. In this case, they are effectively the same, but I've always stuck to `assert` out of habit :)

Reply:
Aah, gotcha. Thanks for explaining 🙂